src/extension.ts

```diff
@@ -47,6 +49,69 @@ export function activate(context: ExtensionContext) {
     })
   );
 
+  // When extension is started, loop for each text documents which are opened to update their language ID.
+  workspace.textDocuments.forEach(document => {
+    updateLanguageId(document);
+  });
+  // When a text document is opened,  update their language ID.
+  context.subscriptions.push(
+    workspace.onDidOpenTextDocument((document) => {
```

Comment:
check for quarkus project here

Reply:
As I add a comment after, checking quarkus project can take some times, so we must do that at then end.

Reply:
right, we should make sure we're dealing with potential problematic files first